### PR TITLE
Initial Docker image and Kubernetes manifests

### PR DESCRIPTION
**Pull request by snormore**
CON-1516

### CODE REVIEW

File: Dockerfile
Comment (nanzhong) [inline Dockerfile]: TIL, is this the new preferred way to install docker on ubuntu? I always thought getting it from docker's repository was the recommended approach.

File: Dockerfile
Comment (nanzhong) [inline Dockerfile]: Could we pin this to a specific version?

File: k8s/deployment.yaml
Comment (nanzhong) [inline k8s/deployment.yaml]: Should we also include the same volume mounts in the deployment version? I can see how it feels less relevant since we aren't targeting a host, but it might be useful to sanity check that our worker node image bootstrapped correctly.

File: k8s/daemonset.yaml
Comment (nanzhong) [inline k8s/daemonset.yaml]: What are you thought on also including a few directories that are specific to our images? In particular `/etc/etcd` and `/opt/k8saas`?

File: script/env
Comment (nanzhong) [inline script/env]: Was this leftover from debug? :P

**Review by nanzhong — APPROVED** [review]
lgtm after changing the docker repo to digitalocean.

File: README.md
Comment (timoreimann) [inline README.md]: The README could provide a high-level overview of what kind of tools the image contains and how it can be used.

Maybe also mention on which Docker Hub repo it is hosted and how it can be pulled.

Could either be done within this PR or a follow up.

File: k8s/daemonset.yaml
Comment (timoreimann) [inline k8s/daemonset.yaml]: Should this one say `digitalocean/doks-debug`?

**Review by timoreimann — CHANGES_REQUESTED** [review]
Marking as "request changes" just to be sure -- I assume the Docker Hub repo should say `digitalocean`?

File: k8s/daemonset.yaml
Comment (snormore) [reply k8s/daemonset.yaml]: Yea still waiting on access for that. Won't merge until it's sorted out.

**Review by timoreimann — APPROVED** [review]

File: README.md
Comment (fatih) [reply README.md]: I would like to see a more documentation. How do we built this image? Why do we need this image? What's the main goal of this repo? Is it for users of DOKS or for us? etc...

Comment (snormore) [reply README.md]: https://github.com/digitalocean/doks-debug/blob/initial/README.md 👀 🙏

**Pull request merged**
